### ISSUE TITLE
Is perform_sinkhorn only applied for 2D tensor?

### ISSUE THREAD

**Issue by daifeng2016** [csyanbin/SCOT#5]
Hi, when reading the code, I found that "torch.mm(K.t(),a)" is used in perform_sinkhorn function. Does that mean only 2D correlation tensor can be processed per time. But in most cases, corretation tensor is 3D or 4D tensor.

**Comment by csyanbin** [comment]
Yes, as far as I know, the original Sinkhorn only supports 2D correlation calculation.
